backend/prompt_studio/prompt_studio_registry/prompt_studio_registry_helper.py

```diff
@@ -139,9 +139,20 @@ def update_or_create_psr_tool(
                 logger.info(f"PSR {obj.prompt_registry_id} was created")
             else:
                 logger.info(f"PSR {obj.prompt_registry_id} was updated")
+
             obj.shared_to_org = shared_with_org
-            obj.shared_users.clear()
-            obj.shared_users.add(*user_ids)
+            if not shared_with_org:
```

Comment:
```suggestion
            obj.shared_users.clear()
            if not shared_with_org:
                obj.shared_users.add(*user_ids)
                # add prompt studio users
                # for shared_user in custom_tool.shared_users:
                obj.shared_users.add(
                    *custom_tool.shared_users.all().values_list("id", flat=True)
                )
                # add prompt studio owner
                obj.shared_users.add(custom_tool.created_by)
```

Reply:
@johnyrahul Can this be refactored as mentioned suggestion, by this way we could avoid an else statement moving clear() method at top. Please check.